### DISCUSSION TITLE
intermittent connection refused on 6443

### DISCUSSION THREAD

**Discussion by abbyers73** [k3s-io/k3s#8263] [Q&A] [answered]
**Environmental Info:**
K3s Version: 
WARN[0000] Failed to resolve user home directory: getwd: no such file or directory
k3s version v1.27.4+k3s1 (36645e73)
go version go1.20.6

Node(s) CPU architecture, OS, and Version: 
Linux <servername> 5.15.0-1035-raspi #38-Ubuntu SMP PREEMPT Tue Jul 18 12:17:11 UTC 2023 aarch64 aarch64 aarch64 GNU/Linux

Cluster Configuration:
right now only the control-plane/master is installed with a external postgresDB

**Describe the bug:**
intermittently the below error is received when running "k3s kubectl get nodes"
The connection to the server 127.0.0.1:6443 was refused - did you specify the right host or port?

**Steps To Reproduce:**
completely removed k3s
ran install with : curl -sfL https://get.k3s.io | sh -s - server --datastore-endpoint="postgres://<DB connection info>"

execute k3s kubectl get nodes

**Expected behavior:**
NAME         STATUS   ROLES                  AGE   VERSION
<servername>   Ready    control-plane,master   11m   v1.27.4+k3s1

**Actual behavior:**
The connection to the server 127.0.0.1:6443 was refused - did you specify the right host or port?

**Additional context / logs:**

**Comment by brandond** [comment]
Sounds like it's crashing. Have you checked the server logs?

if the postgres database is also running on the Pi, you probably don't have enough CPU or disk IO to support that database engine. This would be indicated by latency warnings in the k3s service log, followed by the service crashing - usually due to failure to renew leases in a timely manner.

**Comment by abbyers73** [comment, marked as answer]
the issue was a missing apt package on all the nodes:

resolved by 👍 apt install linux-modules-extra-raspi

**Reply by brandond** [reply]
Ah, that would also do it!
